core/src/string/interner.rs

```diff
@@ -53,80 +53,99 @@ pub struct AvmStringInterner<'gc> {
     interned: WeakSet<'gc, AvmStringRepr<'gc>>,
 
     empty: Gc<'gc, AvmStringRepr<'gc>>,
-    chars: [Gc<'gc, AvmStringRepr<'gc>>; 128],
+    chars: [Gc<'gc, AvmStringRepr<'gc>>; INTERNED_CHAR_LEN],
 }
 
+const INTERNED_CHAR_LEN: usize = 128;
+static INTERNED_CHARS: [u8; INTERNED_CHAR_LEN] = {
```

Comment:
Can `INTERNED_CHARS` be declared as `const`?

Reply:
Probably? (not 100% sure, depends on how const-promotion works exactly) I do feel that `static` is closer to the intent of the code here.

Reply:
It does work for me if I change it to `const`. I can amend if you'd like (I have no opinions on it, just offering to do the key pressing and mouse clicking).

Reply:
Huh, ok. Still prefer `static` though :D

Reply:
I was skeptical at first as well, but the Rust documentation makes it sounds like with `const` we could get the table duplicated every time it's used?

> Static variables, on the other hand, point to a single location in memory, which all accesses share. This means that, unlike with constants, they can’t have destructors, and act as a single value across the entire codebase.

https://doc.rust-lang.org/std/keyword.const.html

Reply:
Not necessarily the entire table, maybe only the elements one by one.

I've seen something similar in `idct.rs` in the `h263-rs` repo, with `BASIS_TABLE`.

Reply:
Those constants are used in fairly tight number crunching loops, and I believe it helped perf a lot to have them inlined.

Reply:
https://github.com/ruffle-rs/h263-rs/blob/f0fa94c366a1d0383df99aa835add175658d6bad/h263/src/decoder/cpu/idct.rs#L36-L65

https://godbolt.org/z/aETTWoncv

Reply:
TBF, in this example, changing `const` to `static` makes no difference whatsoever - I guess the optimizer passes know their stuff. (And this table never has its address taken or anything.)

Reply:
Makes sense. `static` is okay here then.